src/pages/Lending/hooks/useLendingQuoteQuery.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  BASE_BPS_POINTS,
+  fromThorBaseUnit,
+} from 'state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
+import {
+  selectFirstAccountIdByChainId,
+  selectPortfolioAccountMetadataByAccountId,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type UseLendingQuoteQueryProps = {
+  collateralAssetId: AssetId
+  borrowAssetId: AssetId
+  depositAmountCryptoPrecision: string
+}
+export const useLendingQuoteQuery = ({
+  collateralAssetId,
+  borrowAssetId,
+  depositAmountCryptoPrecision,
+}: UseLendingQuoteQueryProps) => {
+  const [borrowAssetReceiveAddress, setBorrowAssetReceiveAddress] = useState<string | null>(null)
+
+  const wallet = useWallet().state.wallet
+  // TODO(gomes): programmatic
+  const accountId =
+    useAppSelector(state =>
+      selectFirstAccountIdByChainId(state, fromAssetId(collateralAssetId).chainId),
+    ) ?? ''
+
+  // TODO(gomes): programmatic
+  const destinationAccountId =
+    useAppSelector(state =>
+      selectFirstAccountIdByChainId(state, fromAssetId(borrowAssetId).chainId),
+    ) ?? ''
+  const destinationAccountMetadataFilter = useMemo(
+    () => ({ accountId: destinationAccountId }),
+    [destinationAccountId],
+  )
+  const destinationAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, destinationAccountMetadataFilter),
+  )
+  const collateralAsset = useAppSelector(state => selectAssetById(state, collateralAssetId))
+  const collateralAssetMarketData = useAppSelector(state =>
+    selectMarketDataById(state, collateralAssetId),
+  )
+
+  const borrowAsset = useAppSelector(state => selectAssetById(state, borrowAssetId))
+  const borrowAssetMarketData = useAppSelector(state => selectMarketDataById(state, borrowAssetId))
+
+  const getBorrowAssetReceiveAddress = useCallback(async () => {
+    if (!wallet || !destinationAccountId || !destinationAccountMetadata || !borrowAsset) return
+
+    const deviceId = await wallet.getDeviceID()
+    const pubKey = isLedger(wallet) ? fromAccountId(destinationAccountId).account : undefined
+
+    return getReceiveAddress({
+      asset: borrowAsset,
+      wallet,
+      deviceId,
+      accountMetadata: destinationAccountMetadata,
+      pubKey,
+    })
+  }, [borrowAsset, destinationAccountId, destinationAccountMetadata, wallet])
+
+  useEffect(() => {
+    ;(async () => {
+      const address = await getBorrowAssetReceiveAddress()
+      if (address) setBorrowAssetReceiveAddress(address)
+    })()
+  }, [getBorrowAssetReceiveAddress])
+
+  const lendingQuoteQueryKey = useDebounce(
+    () => [
+      'lendingQuoteQuery',
+      {
+        borrowAssetReceiveAddress,
+        collateralAssetId,
+        borrowAssetId,
+        depositAmountCryptoPrecision,
+      },
+    ],
+    500,
+  ) as unknown as [string, UseLendingQuoteQueryProps & { borrowAssetReceiveAddress: string }]
```

Comment:
Does this break because `useDebounce` doesn't infer types correctly?

Reply:
Correct, it thinks it's a function where it's actually a function *call*, didn't want to spend too much time on this, and confirmed by introspecting all queries the keys are what we'd expect